src/Microsoft.Net.Http.Headers/NameValueHeaderValue.cs

```diff
@@ -21,6 +21,9 @@ public class NameValueHeaderValue
 
         private StringSegment _name;
         private StringSegment _value;
+        private bool _quotedValue;
```

Comment:
It a decision between whether we should **always add quotes** around the value in SetAndEncodeValue or do it based on whether the value **originally had quotes** around it when parsed or constructed. I decided on the latter.